test/NuGet.Core.Tests/NuGet.Configuration.Test/SettingsFileParsingTests/SourceItemTests.cs

```diff
@@ -76,18 +83,58 @@ public void SourceItem_ParsedSuccessfully()
                 var section = settingsFile.GetSection("packageSources");
                 section.Should().NotBeNull();
 
-                var children = section.Items.ToList();
+                var children = section.Items.Select(c => c as SourceItem).ToList();
```

Comment:
[please don't use `as` to cast, unless you handle nulls](https://github.com/NuGet/NuGet.Client/pull/5309) 😭 

```suggestion
                var children = section.Items.Cast<SourceItem>().ToList();
```

Reply:
Thanks! Fixed.